src/content/drafts/how-to-create-a-custom-eslint-rule.md

```diff
@@ -0,0 +1,282 @@
+* The `recommended` key (line 6) is either `true` or `false` depending on if it should be a default for anyone using the recommended ESLint rules. In the bacon example, I set `recommended` to `false` because this rule should not be a default rule for anyone using ESLint.
+* The `schema` key (line 10) allows you to add options to the rule so that people can configure it in different ways. In the bacon example, I left `schema` as `[]` because there are no options to configure. An example of something that could be added in the future would be flags like `allowBaconInClassNames` or `preventUnderscoredBacon` -- which could then change what the rule does.
+* The `messages` key (line 12) is a way of providing one or more warnings / errors for the user. By defining these here, it makes it so you don't have to copy the same message to many different files or locations. For instance, the tests I wrote above only have to reference the key of `avoidBacon` instead of copying the entire message to that file.
+* In the `avoidBacon` message (line 13), you can see that I have `identifier` surrounded by double brackets. By enclosing a variable name in double brackets like this, I'm telling ESLint that this warning message expects a variable called `identifier` to be passed in. I'll need to add that in the context section below. You can use this to provide better error messages to your users.
+
+If you want to learn more about what flags and settings exist, please read the  [ESLint Rule Basics](https://eslint.org/docs/latest/developer-guide/working-with-rules#rule-basics) guide.
+
+#### Context Object
+
+For the bacon example, the `context` object would look like: 
+
+```javascript
+create(context) {
+  return {
+    Identifier(node) {
+      const identifier = node.name;
+
+      if (identifier.toLowerCase().includes("bacon")) {
+        context.report({
+          node: node,
+          messageId: 'avoidBacon',
+          data: {
+            identifier
+          }
+        });
+      }
+    }
+  };
+},
+```
+
+All rules need to have a `create(context)` in them. What's inside of there, though, is where the differences lie. Let's talk through a few key pieces:
+
+* The `create(context)` object is an object that ESLint uses when it "visits" each of the nodes while traversing the tree. The simplest rules, like our bacon one, specify a specific type of node and then run some if statements to check whether or not the rule is violated. More advanced rules will provide many different types of nodes and functions in combination.
+  * Remember earlier how you used the AST Explorer to get a better understanding of how code is broken down? You'll now put that to use here by typing something like: `Identifier(node) {}` and then adding some code for parsing whatever object you went with inside of the brackets.
```

Comment:
I'm not quite sure what you're saying here. "Creating something like" is a bit confusing because we are using a specific example in the post. Also the phrase "parsing whatever object" is incorrect (you can't parse objects, you can parse strings), so I'm not entirely sure what it is you're trying to say here. Do you mean that you can look at the node that was passed in to determine if your rule passes?

Reply:
Apologies for the confusion. My intention was to call out that the `Identifier(node) {}` is something that would be changed depending on the rule. So if someone was wanting a rule about functions, they wouldn't use the `Identifier` part of things.

I'll re-write the section to try and make it clearer.